lib/autoload/preformatted_text_line_numbers_filter.rb

```diff
@@ -3,6 +3,11 @@ class PreformattedTextLineNumbersFilter < HTML::Pipeline::Filter
   # The regex for splitting input by newlines.
   NEWLINE_REGEX = /\r\n|\r|\n/.freeze
 
+  def initialize(doc, line_start = 1, context = nil, result = nil)
+    @line_start = line_start
```

Comment:
I adjust the start line number outside, and it's no harm even users pass in negative values.